src/BigQuery/Table.php

```diff
@@ -0,0 +1,339 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery;
+
+use Google\Cloud\BigQuery\Connection\ConnectionInterface;
+use Google\Cloud\Storage\Object;
+
+/**
+ * [BigQuery Tables](https://cloud.google.com/bigquery/docs/tables) are a
+ * standard two-dimensional table with individual records organized in rows, and
+ * a data type assigned to each column (also called a field).
+ */
+class Table
+{
+    use JobConfigurationTrait;
+
+    /**
+     * @var ConnectionInterface $connection Represents a connection to BigQuery.
+     */
+    private $connection;
+
+    /**
+     * @var array The table's metadata
+     */
+    private $data;
+
+    /**
+     * @var array The table's identity.
+     */
+    private $identity;
+
+    /**
+     * @param ConnectionInterface $connection Represents a connection to
+     *        BigQuery.
+     * @param string $id The table's id.
+     * @param string $datasetId The dataset's id.
+     * @param string $projectId The project's id.
+     * @param array $data The table's metadata.
+     */
+    public function __construct(ConnectionInterface $connection, $id, $datasetId, $projectId, array $data = [])
+    {
+        $this->connection = $connection;
+        $this->data = $data;
+        $this->identity = [
+            'tableId' => $id,
+            'datasetId' => $datasetId,
+            'projectId' => $projectId
+        ];
+    }
+
+    /**
+     * Check whether or not the table exists.
+     *
+     * Example:
+     * ```
+     * $table->exists();
+     * ```
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        try {
+            $this->connection->getTable($this->identity + ['fields' => 'tableReference']);
+        } catch (\Exception $ex) {
```

Comment:
Sounds fair. My only concern is that it swallows everything, so it is possible that suddenly `exists()` cease to work in a very mysterious manner.